Runs: semantic run colors (#1407)
wandb/docs: content/en/guides/models/track/runs/color-code-runs.md

---
description: Create semantic legends for charts
menu:
  default:
    identifier: color-code-runs
    parent: what-are-runs
title: Semantic run plot legends
---

Create meaningful legends for run line plots to visually distinguish them based on metrics you log to W&B. Identify trends and patterns based on the highest, lowest, or latest values of a metric across your training and evaluation runs. Runs are bucketed and colored in plots based on the plot's configured metric, Y value, and number of buckets. 

Navigate to your workspace's settings page to configure the metric-based colors for runs:

1. Navigate to your W&B project.
2. Select the **Workspace** tab from the project sidebar.
3. Click on the **Settings** icon (⚙️) in the top right corner.
4. From the drawer, select **Runs** then select **Metric-based colors**.
    - From the **Metric** dropdown, select the metric you want to use for assigning colors to runs.
    - From the **Y value** dropdown, select the y value you want to use for assigning colors to runs.

>     - Set the the number of buckets to a value from 2 to 8.

Reviewer: ```suggestion
    - Set **Number of buckets** to a value from 2 to 8. Optionally, click the color picker for a bucket to customize its color.
```
I can't tell if this one didn't get submitted properly in my previous review or if you decided not to use it. If the latter, ignore this one too. :)

---
description: Group training and evaluation runs into larger experiments
menu:
  default:
    identifier: color-code-runs
    parent: what-are-runs
title: Color code runs
---

> Color code runs to visually distinguish them based on metrics you log to W&B. The metric, y value, and the number of buckets you select determine how the runs are grouped and colored in a plot. 

Reviewer: The description doesn't seem to match the title or the intro -- this is more specifically about colors than the more general concept of grouping, right?

> <!-- This is useful for quickly identifying runs that meet certain criteria, such as the latest, highest, or lowest values of a metric. -->

Reviewer: This was a good call to action. Can we replace it with any other "why"?

> To color code runs by a metric, you can set the metric-based colors in your W&B workspace settings.

Reviewer: "color code" looks funny to me as a verb. "Set metric-based colors for runs" in line 16 is better. WDYT?

To set metric-based colors for runs in your W&B project:

1. Navigate to your W&B project.
2. Select the **Workspace** tab from the project sidebar.

> 3. Click on the **Settings** icon (⚙️) in the top right corner of the workspace.

Reviewer: ```suggestion
3. Click the **Settings** icon (⚙️) in the top right corner.
```
Redundant "of the workspace" given mentions on line 14, 18, 19.

4. From the drawer, select **Runs**.
5. Select **Metric-based colors**.
6. From the **Metric** dropdown, select the metric you want to use for color coding.

> 7. From the **Y value** dropdown, select the y value you want to use for color coding.

Reviewer: ```suggestion
4. From the drawer, select **Runs** then select **Metric-based colors**.
    - From the **Metric** dropdown, select the metric you want to use for color coding.
    - From the **Y value** dropdown, select the y value you want to use for color coding.
```
WDYT? This reduces the # of steps so that the task doesn't seem as complex. If you agree, then line 25 becomes 5 instead of 8.

8. Select the number of buckets. Minimum is 2, maximum is 8.

The following sections describe how to set the metric and y value for color coding runs, as well as how to customize the buckets used for color coding.

> ## Configure metric

Reviewer: Does this mean "Select the metric"? The current heading is a bit terse.

> The options in your **Metric** dropdown are derived from the key-value pairs you log to W&B and default metrics provided by W&B.

Reviewer: Is "and default metrics...W&B" a fragment? I see the subsection I think it refers to on line 33, but I'm not 100% sure.

You can choose from the following options:

- **Latest**: Determine color based on Y value at last logged step for each line.
- **Max**: Color based on highest Y value logged against the metric.
- **Min**: Color based on lowest Y value logged against the metric.

## Customize buckets

Buckets are ranges of values that determine the color of a run. Runs that fall within that bucket's range are displayed in that color. Each bucket is assigned a unique color, and the buckets are evenly distributed across the range of values for the specified metric.

Consider the following configuration:

{{< img src="/images/track/color-coding-runs.png" alt="" >}}

**Metric** is set to `"Accuracy"` (abbreviated as `"acc"`). **Y value** is set to `"Max"`, meaning the maximum value of the accuracy metric is used to determine the color coding for the runs. 

Six buckets are defined for the metric, with each bucket representing a range of accuracy values. Within the **Buckets** section, the following range of buckets are defined:

- Bucket 1: (Min - 0.5798)
- Bucket 2: (0.5798 - 0.6629)
- Bucket 3: (0.6629 - 0.7460)
- Bucket 4: (0.7460 - 0.8292)
- Bucket 5: (0.8292 - 0.9123)
- Bucket 6: (0.9123 - Max)

> In the line plot below, the run with the highest accuracy (0.957) is colored in a deep purple (Bucket 6), while the run with the lowest accuracy (0.7993) is colored in a lighter purple (Bucket 4). The other runs are colored based on their accuracy values, with the color gradient indicating their relative performance. 

Reviewer: It's a shame that the plot image only shows purples, when the config image you showed has oranges and pinks. Is there a way to show all of the colors or at least more of them?

Author: How bout this? We lose the purple but we see (sort of) more colors:

<img width="631" alt="Screenshot 2025-06-23 at 12 44 25 PM" src="https://github.com/user-attachments/assets/3638bb40-e976-4c9b-92c9-515bbc24d04f" />

Reviewer: This is much better. Even better if you shift all the colors a bit darker / more saturated, for readability. These are a bit light. (nit)

Author: I don't have that option :/

Author: (The UI has hardcoded colors.)

Author: Hmm...so previewing it with the new image in hugo makes it look extra bright..For now, the original will have to do.

---
description: Create semantic legends for charts
menu:
  default:
    identifier: color-code-runs
    parent: what-are-runs
title: Semantic run plot legends
---

> Create meaningful legends for run line plots to visually distinguish them based on metrics you log to W&B. Identify trends and patterns based highest, lowest, or latest values of a metric across your training and evaluation runs. The metric, y value, and the number of buckets you select determine how the runs are bucketed and colored in a plot. 

Reviewer: ```suggestion
Create meaningful legends for run line plots to visually distinguish them based on metrics you log to W&B. Identify trends and patterns based on the highest, lowest, or newest values of a metric across your training and evaluation runs. Runs are bucketed and colored in plots based on the plot's configured metric, Y-axis value, and number of buckets. 
```

Author: In the UI they use "latest" instead of "newest" and "y value" instead of "y-axis"

Navigate to your workspace's settings page to configure the metric-based colors for runs:

1. Navigate to your W&B project.
2. Select the **Workspace** tab from the project sidebar.
3. Click on the **Settings** icon (⚙️) in the top right corner.
4. From the drawer, select **Runs** then select **Metric-based colors**.
    - From the **Metric** dropdown, select the metric you want to use for assigning colors to runs.
    - From the **Y value** dropdown, select the y value you want to use for assigning colors to runs.

>     - Select the number of buckets. Minimum is 2, maximum is 8.

Reviewer: ```suggestion
    - Set **Number of buckets** to a value from 2 to 8. Optionally, customize each bucket's color using the picker.
```

> The following sections describe how to set the metric and y value, as well as how to customize the buckets used for assigning colors to runs.

Reviewer: ```suggestion
The following sections describe how to set the metric and Y-axis value and customize the buckets used for assigning colors to runs.
```

Author: Oddly enough it's called "y value" not "y-axis".

### Custom metrics

> Use custom metrics that you log to W&B in your training or evaluation scripts to color code runs and create meaningful plot legends. Custom metrics are logged as key-value pairs, where the key is the name of the metric and the value is the metric value.

Reviewer: ```suggestion
Color runs and create meaningful plot legends based on custom metrics logged by your training or evaluation scripts. Custom metrics are logged as key-value pairs, where the key is the name of the metric and the value is the metric value.
```

{{< img src="/images/track/color-coding-runs.png" alt="" >}}

> **Metric** is set to `"Accuracy"` (abbreviated as `"acc"`). **Y value** is set to `"Max"`, meaning the maximum value of the accuracy metric is used to determine the color of a run.

Reviewer: ```suggestion
- **Metric** is set to `"acc"`, which represents accuracy.
- **Y value** is set to `"Max"`, which determines the color of a run based on the maximum accuracy logged during the run.
```
It is confusing to mark up `"Accuracy"` when that's not what the metric is called.
